mindmeld/auto_annotator.py

```diff
@@ -759,4 +773,67 @@ def _resolve_person(self, entity):
         return entity
 
 
+class BootstrapAnnotator(Annotator):
+    """ Bootstrap Annotator class used to generate annotations based on existing annotations.
+    """
+
+    def __init__(self, app_path, config=None):
+        super().__init__(app_path=app_path, config=config)
+        self.confidence_threshold = self.config.get("confidence_threshold", 0)
+        logger.info("BootstrapAnnotator is loading %s.", self.app_path)
+        self.nlp = NaturalLanguageProcessor(self.app_path)
+        self.nlp.build()
+
+    def parse(self, sentence, entity_types, domain: str, intent: str, **kwargs):
+        """
+        Args:
+                sentence (str): Sentence to detect entities.
+                entity_types (list): List of entity types to parse. If None, all
+                        possible entity types will be parsed.
+        Returns: entities (list): List of entity dictionaries.
+        """
+        er = self.nlp.domains[domain].intents[intent].entity_recognizer
+        entities = []
+        for index, (entity, confidence) in enumerate(er.predict_proba(sentence)):
+            if (
+                not entity_types or entity.entity.type in entity_types
+            ) and confidence >= self.confidence_threshold:
+
+                role = self.nlp.process(
+                    sentence, allowed_nlp_classes={domain: {intent: {}}}
+                )["entities"][index]["role"]
```

Comment:
you could just call `self.nlp.domains[domain].intents[intent].entities[entity.type].role_classifier.predict()`, though I also wonder here if it would be noticeably slower to just use `self.nlp.process(sentence, verbose=True)` on L797.

Reply:
Refactored the Parse function to only need to use `self.nlp.process(sentence, verbose=True)`.  There hasn't been any noticeable decline in speed.